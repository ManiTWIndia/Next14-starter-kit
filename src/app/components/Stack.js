import PropTypes from 'prop-types';

function Stack({
    children, spacing = 2, direction = 'row', wrap = false
}) {
    const style = {
        display: 'flex',
        gap: `${spacing * 0.25}rem`,
        flexWrap: wrap ? 'wrap' : 'nowrap',
        flexDirection: direction
    };

    return <div style={style}>{children}</div>;
}

Stack.propTypes = {
    children: PropTypes.node,
    spacing: PropTypes.number,
    wrap: PropTypes.bool,
    direction: PropTypes.oneOf(['row', 'column'])
};

Stack.defaultProps = {
    children: undefined,
    spacing: 2,
    direction: 'row',
    wrap: false
};

export default Stack;
