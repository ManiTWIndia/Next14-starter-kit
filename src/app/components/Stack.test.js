import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';
import Stack from './Stack';

describe('Stack', () => {
    it('renders a stack', () => {
        const {container} = render(
            <Stack>
                {[...Array(4).keys()].map((n) => (
                    <div
                        key={n}
                        style={{
                            width: '50px',
                            height: '50px',
                            backgroundColor: 'red',
                            display: 'flex',
                            justifyContent: 'center',
                            alignItems: 'center'
                        }}
                    >
                        {n + 1}
                    </div>
                ))}
            </Stack>
        );
        expect(container).toMatchSnapshot();
    });

    it('should trigger onClick on click of the button', () => {
        render(
            <Stack>
                {[...Array(2).keys()].map((n) => (
                    <div
                        key={n}
                        data-testid={`component-${n + 1}`}
                        style={{
                            width: '50px',
                            height: '50px',
                            backgroundColor: 'red',
                            display: 'flex',
                            justifyContent: 'center',
                            alignItems: 'center'
                        }}
                    >
                        {n + 1}
                    </div>
                ))}
            </Stack>
        );
        expect(screen.getByTestId('component-1')).toBeInTheDocument();
        expect(screen.getByTestId('component-2')).toBeInTheDocument();
    });
});
