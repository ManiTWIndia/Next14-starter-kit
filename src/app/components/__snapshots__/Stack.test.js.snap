// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack renders a stack 1`] = `
<div>
  <div
    style="display: flex; gap: 0.5rem; flex-wrap: nowrap; flex-direction: row;"
  >
    <div
      style="width: 50px; height: 50px; background-color: red; display: flex; justify-content: center; align-items: center;"
    >
      1
    </div>
    <div
      style="width: 50px; height: 50px; background-color: red; display: flex; justify-content: center; align-items: center;"
    >
      2
    </div>
    <div
      style="width: 50px; height: 50px; background-color: red; display: flex; justify-content: center; align-items: center;"
    >
      3
    </div>
    <div
      style="width: 50px; height: 50px; background-color: red; display: flex; justify-content: center; align-items: center;"
    >
      4
    </div>
  </div>
</div>
`;
